bciers/apps/registration/tests/components/operations/registration/OperationInformationForm.test.tsx

```diff
@@ -23,15 +112,181 @@ describe("the OperationInformationForm component", () => {
+      "/register-an-operation/uuid2/2?title=Operation 2",
+    );
+  });
+  it("should submit when a purpose with regulated products is selected", async () => {
+    actionHandler.mockReturnValueOnce({ id: "uuid2", name: "Operation 2" });
+    render(
+      <OperationInformationForm
+        formData={{}}
+        schema={localSchema}
+        step={1}
+        steps={allOperationRegistrationSteps}
+      />,
+    );
+
+    const purposeInput = screen.getByRole("combobox", {
+      name: /The purpose of this registration+/i,
+    });
+
+    const openPurposeDropdownButton = purposeInput?.parentElement?.children[1]
+      ?.children[0] as HTMLInputElement;
+
+    await userEvent.click(openPurposeDropdownButton);
+    const purposeOption = screen.getByText("OBPS Regulated Operation");
+    await userEvent.click(purposeOption);
+
+    const regulatedProductsInput = screen.getByPlaceholderText(
+      /select regulated product.../i,
+    );
+
+    const openProductsDropdown = regulatedProductsInput?.parentElement
+      ?.children[1]?.children[0] as HTMLInputElement;
+    await userEvent.click(openProductsDropdown);
+    const product1 = screen.getByText(
+      "BC-specific refinery complexity throughput",
+    );
+    await userEvent.click(product1);
+    await userEvent.click(openProductsDropdown);
+    const product2 = screen.getByText("Cement equivalent");
+    await userEvent.click(product2);
+
+    const operationInput = screen.getByRole("combobox", {
+      name: /Select your operation+/i,
+    });
+    const openOperationsDropdownButton = operationInput?.parentElement
+      ?.children[1]?.children[0] as HTMLInputElement;
+    await userEvent.click(openOperationsDropdownButton);
+    const operationOption = screen.getByText(/Operation 2/i);
+    await userEvent.click(operationOption);
+
+    userEvent.click(screen.getByRole("button", { name: /save and continue/i }));
+    await waitFor(() => {
+      expect(actionHandler).toHaveBeenCalledWith(
+        "registration/v2/operations/uuid2/1",
+        "PUT",
+        "",
+        {
+          body: JSON.stringify({
+            registration_purpose: "OBPS Regulated Operation",
+            operation: "uuid2",
+            regulated_products: [1, 2],
+          }),
+        },
+      );
+    });
+    expect(mockPush).toHaveBeenCalledWith(
+      "/register-an-operation/uuid2/2?title=Operation 2",
+    );
+  });
+  it("should trigger required field errors when a purpose without regulated products", async () => {
+    render(
+      <OperationInformationForm
+        formData={{}}
+        schema={localSchema}
+        step={1}
+        steps={allOperationRegistrationSteps}
+      />,
+    );
+    userEvent.click(screen.getByRole("button", { name: /save and continue/i }));
+    await waitFor(() => {
+      expect(screen.getAllByText(/Required field/i)).toHaveLength(2);
+    });
+  });
+  it("should trigger required field errors with regulated products", async () => {
```

Comment:
This test is currently failing because the `required` in the schema isn't working

Reply:
Commented out and noted in the multistep bug ticket